core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -416,6 +428,7 @@ public void insertRecord(Object recordBase, long recordOffset, int length, int p
     Platform.copyMemory(recordBase, recordOffset, base, pageCursor, length);
     pageCursor += length;
     inMemSorter.insertRecord(recordAddress, partitionId);
+    inMemRecordsSize += length;
```

Comment:
Should we also include the uaoSize?

Reply:
+1, the `pageCursor` is also increased by uaoSize and length